shared/health/src/main/java/io/pravega/shared/health/Health.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.shared.health;
+
+import com.google.common.collect.ImmutableMap;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.util.Map;
+
+/**
+ * The {@link Health} class represents the data gathered by a {@link HealthContributor} after performing a health check.
+ */
+@Builder
+@ThreadSafe
+@AllArgsConstructor
+public class Health {
+
+    /**
+     * Any sort of identifying string that describes from which component this measurement
+     * was taken from.
+     */
+    @Getter
+    private final String name;
+
+    @Getter
+    @Builder.Default
+    private final Status status = Status.UNKNOWN;
+
+    @Getter
+    @Builder.Default
+    private final Map<String, Object> details = ImmutableMap.of();
```

Comment:
What is this map with _Object_ values for? How does it help provide health information?

Reply:
This is for any information that a `HealthContributor` wants to expose, apart from its health `Status`. The reason why its of type `Object` is to allow the consuming service whether it be the controller or segementstore to define how to how these objects should be serialized.

Reply:
Ok.